### ISSUE TITLE
Bunch of encodings are missing

### ISSUE THREAD

**Issue by Kojoley** [ikvmnet/ikvm#329]
cp | jdk8 | ikvm
----|----|-------
437 | ok | ok
850 | ok | missing
852 | ok | missing
855 | ok | missing
857 | ok | missing
860 | ok | missing
861 | ok | missing
863 | ok | missing
865 | ok | missing
866 | ok | missing
869 | ok | missing
936 | ok | ok
65001 | missing | missing

This makes applications that use log4j to output to console unusable.

```java
public final class Main
{
    public static void main(String[] args)
    {
        java.nio.charset.Charset.forName(System.getProperty("sun.stdout.encoding"));
    }
}
```

```
C:\w>chcp 850
850

C:\w>"C:\Program Files\Java\jdk1.8.0_301\bin\javac.exe" Main.java 

C:\w>"C:\Program Files\Java\jdk1.8.0_301\bin\java.exe" Main

C:\w>"G:\ikvm\bin\java.exe" Main
Exception in thread "main" java.nio.charset.UnsupportedCharsetException: ms850
	at java.nio.charset.Charset.forName(Charset.java:531)
	at Main.main(Main.java:34)
	at java.lang.reflect.Method.invoke(Method.java:486)
	at IKVM.Java.Externs.ikvm.runtime.Launcher.run(Unknown Source)
```

**Comment by wasabii** [comment]
Hmm. Thanks for this. Thought I had a test case covering those. Will address.

**Comment by Kojoley** [comment]
> So, I added some tests to check all of these. They all worked for me except 65001. Which verison of IKVM are you using here?

https://github.com/ikvmnet/ikvm/releases/download/8.5.0/IKVM-8.5.0-image-net461-win7-x64.zip

**Comment by wasabii** [comment]
Got it.

**Comment by Kojoley** [comment]
I've build `hotfix/8.5.1` branch to test if the fix works, but I can't run programs with it altogether (while `develop is fine`)
```
C:\w>G:\ikvm\src\java\bin\Release\net461\java.exe Main
Error: Could not find or load main class Main
```

It doesn't seem to fix `cp65001` either, I think it should return `UTF-8` like OpenJDK 17 does, replicating JDK8 deficiency isn't a right choice here.
```java
public final class Main
{
    public static void main(String[] args)
    {
        System.out.println(System.getProperty("sun.stdout.encoding"));
        java.nio.charset.Charset.forName(System.getProperty("sun.stdout.encoding"));
    }
}
```
```
C:\w>"C:\Program Files\Java\jdk-17.0.5\bin\java.exe" Main
UTF-8
```

**Comment by wasabii** [comment]
Forgot about cp65001. I have to think about this more, as I don't even see that encoding in the source anywhere.

**Comment by Kojoley** [comment]
> Forgot about cp65001. I have to think about this more, as I don't even see that encoding in the source anywhere.

https://github.com/openjdk/jdk11u/blob/31396660db3eae3977bbbf8b978b581dfd941d2d/src/java.base/windows/native/libjava/java_props_md.c#L140-L155

**Comment by wasabii** [comment]
Yeah.... That's JDK11 though.

**Comment by wasabii** [comment]
Is it a thing that exists in later JDK8 versions?

**Comment by AliveDevil** [comment]
No hint for 65001.

https://github.com/openjdk/jdk8u/blob/master/jdk/src/windows/native/java/lang/java_props_md.c#L138-L151

**Comment by wasabii** [comment]
Yeah. This is going to seem annoying to you, then. But I believe we should seek to exactly mirror how OpenJDK 8 works. Working around the issue in IKVM *should* be exactly the same as working around it in any other JDK8 distribution. So, documentation that says do X in Java, applies to us as well.

Here's a more pragmatic reason:

We are getting close to being able to replace much of our forked C# code with the original OpenJDK C/C++ code. If we deviate from OpenJDK here, that would block us from porting the same deviation to the new native version without forking the C code in the future. And that's exactly what we want to avoid.

I don't know the background on this particular issue, so I can't really speak to it. But in general, I would say, if there is a bug in OpenJDK: then we should be fixing it in OpenJDK. OpenJDK is also an open source project. Though their rules for adopting changes are burdensome in comparison to us, it is still an open source project.

If this is a bug that the OpenJDK project doesn't want to fix, I'd imagine we'd adopt their same reasoning. If it is one they would fix, then fixing it there would cause us to eventually benefit from that.

**Comment by wasabii** [comment]
But also, we would of course love help in starting to work on some of the JDK9+ features. :)

**Comment by wasabii** [comment]
We would eventually switch that statement to mirroring JDK9.

**Comment by Kojoley** [comment]
> We would eventually switch that statement to mirroring JDK9 for IKVM9.

Until then what happens with PRs that implement JDK9 features/fixes?

**Comment by wasabii** [comment]
Alright. Well, I think I've got to consider this one closed now, to be released in 8.5.1. Continuing to work on the other reports you made though.
